### ISSUE TITLE
Pass properties to <a> link

### ISSUE THREAD

**Issue by MBO** [frontarm/navi#47]
It would be nice to be able to pass all unrecognized properties form 'NavLink' to generated `<a>` tag, like:

```
<NavLink href="/" data-track="navbar" role="button">
```

and having `data-track` and `role` passed to generated anchor:

```
<a href="/" data-track="navbar" role="button">
```

**Comment by jamesknelson** [comment]
For the record, it's possible to customize the props passed to `<a>` by passing a `render` prop through to `<NavLink>`. Within the `render` function, you can use the `<NavLink.Anchor>` component to render a connected `<a>` tag, and you can pass any props you'd like to `<NavLink.Anchor>`.

For example:

```js
<NavLink href="/" render={props =>
  <NavLink.Anchor data-track="navbar" role="button"/>
} />
```

Because `<NavLink.Anchor>` is effectively just a wrapper around `<a>`, it allows for customization of the underlying `<a>` tag without maintaining  a long list of supported properties. Maybe one day we could include support for the properties in `<NavLink>` itself, but it's something that is rather hard to undo, so I'm hesitant to rush into it.

**Comment by MBO** [comment]
Thanks. The way I did is more reusable in our code - we have lots of those links in app, and each one could have different set of attributes, so it's easier to just strip `NavLink`s properties and only pass rest to `NavLink.Anchor`.

One more question - the way you did it is you're skipping `className`, `style` and `hidden` props, they are not added by `NavLink.Anchor` then, right? If I want to have the same behavior as standard `NavLink` I still have to copy them to my solution?

**Comment by jamesknelson** [comment]
Just published 0.13.0-alpha.0 with `useLinkProps()` and `useActive()` hooks which should make this possible:

- `useLinkProps(props)` accepts the same props as a `<Link>`, and returns an object with `href`, `onClick`, and any other props that you'd want to spread onto an `<a>`.

- `useActive(href)` returns `true` when the current route matches the specified href. You can use this along with `useLinkProps` to create your own custom styled links.

The `<Link>` component now uses these internally, so you can basically create your own custom link component in 2 lines.

**Comment by strass** [comment]
The default Link component isn't passing the `disabled` attribute to the anchor element it renders. I usually style disabled links through that html attribute. Is this intentional? If so, I can use that render option but I thought it might have been missed.
